subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt.c

```diff
@@ -141,16 +141,21 @@ static int img_mgmt_find_tlvs(int slot, size_t *start_off, size_t *end_off, uint
 
 int img_mgmt_active_slot(int image)
 {
-#if CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 2
-	if (image == 1) {
-		return 2;
-	}
-#endif
-	/* Image 0 */
+	int slot = 0;
+
+	/* Multi image does not support DirectXIP currently */
+#if CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER > 1
+	slot = (image << 1);
+#else
+	/* This covers single image, including DirectXiP */
 	if (FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot1_partition)) {
-		return 1;
+		slot = 1;
 	}
-	return 0;
+	slot = 0;
```

Comment:
this will always return 0

Reply:
else seems needed

Reply:
No, that is rebase error, the `slot = 0` should not be here at all.